### PR TITLE
feat: sanitize media filenames according to global slug setting

### PR DESCRIPTION
**Pull request by stefk**
Currently, users are allowed to upload images whose filename contains accents, spaces or special characters. This may cause several problems downstream (https://github.com/netlify/netlify-cms/issues/857). 

It seems that at some point, the work done in #1135 should have fixed that issue (see https://github.com/netlify/netlify-cms/issues/857#issuecomment-369054206) but I can't find any trace of that behaviour in the current codebase.

This PR adds/restores that feature by applying the policy defined in the global `slug` setting to media inserted via the default media library.

### CODE REVIEW

**Comment by erezrokah** [comment]
Thanks @stefk, this does look like a regression.

**Comment by erezrokah** [comment]
Not sure why tests are failing on your branch, I will look into it soon

**Comment by stefk** [comment]
Thanks @erezrokah. I was also puzzled by the test failure, considering everything was green yesterday and that I had just amended a commit this morning with unrelated changes. Then I noticed that the failure was on *editorial_workflow_migration_spec_github_backend.js*, which wasn't present in my branch (it comes from the recent #3316). Rebasing solved the build issue, but do you have any idea how was this situation possible in first place? Why was a spec from the master branch executed on a distinct branch?

**Comment by erezrokah** [comment]
> Thanks @erezrokah. I was also puzzled by the test failure, considering everything was green yesterday and that I had just amended a commit this morning with unrelated changes. Then I noticed that the failure was on _editorial_workflow_migration_spec_github_backend.js_, which wasn't present in my branch (it comes from the recent #3316). Rebasing solved the build issue, but do you have any idea how was this situation possible in first place? Why was a spec from the master branch executed on a distinct branch?

That is a new test and I think it was a bit flaky (hopefully I fixed it).

**Comment by erezrokah** [comment]
> Thanks @erezrokah. I was also puzzled by the test failure, considering everything was green yesterday and that I had just amended a commit this morning with unrelated changes. Then I noticed that the failure was on _editorial_workflow_migration_spec_github_backend.js_, which wasn't present in my branch (it comes from the recent #3316). Rebasing solved the build issue, but do you have any idea how was this situation possible in first place? Why was a spec from the master branch executed on a distinct branch?

That is a new test and I think it was a bit flaky (hopefully I fixed it).

File: packages/netlify-cms-core/src/actions/__tests__/mediaLibrary.spec.js [hunk 1]
```diff
@@ -171,7 +224,6 @@ describe('mediaLibrary', () => {
         });
 
         expect(backend.persistMedia).toHaveBeenCalledTimes(1);
-        expect(backend.persistMedia).toHaveBeenCalledWith(store.getState().config, assetProxy);
```
Comment (erezrokah) [inline packages/netlify-cms-core/src/actions/__tests__/mediaLibrary.spec.js]: What do you think on using:

```js
expect(backend.persistMedia).toHaveBeenCalledWith(
          store.getState().config,
          expect.objectContaining({
            path: 'static/media/name.png',
          }),
        );
```

Comment (stefk) [reply packages/netlify-cms-core/src/actions/__tests__/mediaLibrary.spec.js]: Good idea. I've added that assertion for the sanitization test as well.

**Pull request merged**